File: Java/0001-1000/001-100/11-20/20.java
class Solution {
    public boolean isValid(String s) {
        char[] c = new char[s.length()];
        int i = -1;
        for (char ch : s.toCharArray()) {
            if (ch == '(' || ch == '{' || ch == '[') {
                c[++i] = ch;
            } else if (ch == ')') {
                if (i < 0 || c[i] != '(')
                    return false;
                i--;
            } else if (ch == '}') {
                if (i < 0 || c[i] != '{')
                    return false;
                i--;
            } else {
                if (i < 0 || c[i] != '[')
                    return false;
                i--;
            }
        }
        return i == -1;
    }
}

File: Python/0001-1000/001-100/11-20/20.py
class Solution:
    def isValid(self, s: str) -> bool:
        stk = []
        n = -1
        for c in s:
            if c == '(' or c == '{' or c == '[':
                stk.append(c)
                n += 1
            elif c == ')':
                if n < 0 or stk[n] != '(':
                    return False
                stk.pop()
                n -= 1
            elif c == '}':
                if n < 0 or stk[n] != '{':
                    return False
                stk.pop()
                n -= 1
            elif c == ']':
                if n < 0 or stk[n] != '[':
                    return False
                stk.pop()
                n -= 1
        return n == -1
